src/java/org/apache/cassandra/schema/Schema.java

```diff
@@ -626,7 +628,9 @@ public void resetLocalSchema()
         updateHandler.clear();
 
         logger.debug("Clearing local schema keyspace instances...");
-        clear();
+        distributedKeyspaces.forEach(this::unload);
+        updateVersion(SchemaConstants.emptyVersion);
+        SchemaDiagnostics.schemaCleared(this);
```

Comment:
Those were in a synchronized method before?

Reply:
good point, I need to revisit this

Reply:
Hmm, do we want the whole method now? I mean before it was considered that only those three operations were enough. Just trying to be careful with the price of synchronization but at the same time not to miss it :-)

Reply:
Yes, I think it was a mistake that it wasn't synchronized. Synchronization does not really matter in terms of performance in this case because that method is called only from NodeTool and it is one time action. Basically what we need I think is that there are no other schema changes issued locally when we are restoring schema from the other node, not just during cleaning